Devscord.DiscordFramework/Integration/Services/DiscordClientRolesService.cs

```diff
@@ -68,15 +68,21 @@ public async Task SetRolePermissions(ChannelContext channel, DiscordServerContex
         public async Task SetRolePermissions(IEnumerable<ChannelContext> channels, DiscordServerContext server, ChangedPermissions permissions, UserRole role)
         {
             await Task.Delay(1000);
```

Comment:
po co te Delay(1000)?

Reply:
to dlatego, że po utworzeniu kanału jest jakby jeszcze nie w pełni dostępny, przez co chcąc od razu przypisać uprawnienia dla tego kanału rzucany jest wyjątek
można by to lekko przebudować i pozbyć się Delay
ale myślę, że nie przeszkadza to na tyle że poświęcać temu czas
może kiedyś przy okazji się poprawi

Reply:
trochę to dziwnie wygląda z tym Delay, ale na razie może być, później można to zmienić.

Reply:
no to chyba powinno być na wyższym poziomie a nie podczas ustawiania uprawnień
bo teraz elementy wysokopoziomowe są uzależnione od niskopoziomowych 
tzn 
elementy niskopoziomowe które powinny być reużywalne, nie powinny mieć logiki typowej dla wysokopoziomowych
ten delay powinien być w serwisie który tworzy kanał po czym przydziela mu role 
a nie podczas samego procesu przydzielania roli
bo co jeśli ktoś chciałby użyć tego w innej sytuacji niż od razu po stworzeniu kanału?
np w sytuacji gdzie ten kanał już istnieje od dawna 

może ok opcją byłby if sprawdzający kiedy konkretny kanał został stworzony i jeśli różnica czasu między stworzeniem a aktualnym czasem jest mniejsza niż dajmy na to 1 sekunda -> to czeka tą sekunde
a jak to jest starszy kanał to uznaje że już na pewno cały sie stworzył

=================

i dlaczego to jest w DiscordClient`Roles`Service, skoro ma dotyczyć kanałów?

albo robisz w ChannelsService i mówisz że ustawiasz uprawnienia dla konkretnej roli na kanale
albo robisz dla Roli i mówisz że to ustawienie konkretnej roli na konkretnym kanale

jest jeszcze kwestia tego że role mają uprawnienia w kontekście całego serwera i w kontekście konkretnego kanału 

trzeba to przemyśleć